### PR TITLE
chore: Fix typos

### PR DESCRIPTION
**Pull request by szepeviktor**
## Description

Found many misspellings.

## Change checklist

- [x] Self-review.
- [x] Documentation updates if relevant.
- [x] Tests if relevant.

### CODE REVIEW

File: iroh-net/src/magicsock.rs [hunk 1]
```diff
@@ -1018,7 +1018,7 @@ impl DiscoMessageSource {
 ///   and start a new one when the current one has finished
 #[derive(Debug)]
 struct EndpointUpdateState {
-    /// If running, set to the reason fo the currently the update.
+    /// If running, set to the reason for the currently the update.
```
Comment (szepeviktor) [inline iroh-net/src/magicsock.rs:1021]: This may need DeepL Write or Grammarly.

**Review by flub — APPROVED** [review]

**Comment by flub** [comment]
> Is "entrys" intentional?

"entries" would be the correct spelling

**Review by flub — APPROVED** [review]

**Comment by flub** [comment]
https://github.com/n0-computer/iroh/issues/1966 for flaky test

**Comment by szepeviktor** [comment]
`.typos.toml`
```toml
[files]
extend-exclude = [
    ".git/",
    "iroh-net/proptest-regressions/derp/codec.txt",
]
ignore-hidden = false

[default]
extend-ignore-re = [
    "\\[[0-9a-f]{7}\\]",
    "let ba =",
    "if ab && ba",
    '" worl", "d"',
    "Replace `currrent` with `current`",
]
```

**Comment by flub** [comment]
> `.typos.toml`
> 
> ```toml
> [files]
> extend-exclude = [
>     ".git/",
>     "iroh-net/proptest-regressions/derp/codec.txt",
> ]
> ignore-hidden = false
> 
> [default]
> extend-ignore-re = [
>     "\\[[0-9a-f]{7}\\]",
>     "let ba =",
>     "if ab && ba",
>     '" worl", "d"',
>     "Replace `currrent` with `current`",
> ]
> ```

@szepeviktor want to contribute that as a CI step?  Probably in `ci.yml` after the `cargo_deny` step is a good starting point.

**Comment by szepeviktor** [comment]
It seems like a never-ending story ...

**Comment by flub** [comment]
> It seems like a never-ending story ...

Yes, apologies for doing all this on your PR.  I think I finally found the problem in our CI setup.  I'll probably fix this separately in an other PR and then strip all this noise from this PR and rebase it.

File: .github/workflows/tests.yaml [hunk 2]
```diff
@@ -16,7 +16,7 @@ on:
       git-ref:
         description: 'Which git ref to checkout'
         type: string
-        default: 'main'
+        default: ${{ github.ref }}
```
Comment (szepeviktor) [inline .github/workflows/tests.yaml:19]: This is `github.ref_name`

Comment (szepeviktor) [reply .github/workflows/tests.yaml:19]: You can browse the `github` context in this workflow run output https://github.com/szepeviktor/byte-level-care/actions/runs/7585113229/job/20660246214

Comment (flub) [reply .github/workflows/tests.yaml]: from https://docs.github.com/en/actions/learn-github-actions/contexts i think `github.ref` points to the same and is more descriptive?  Any reason to prefer `ref_name`?

This is moved to https://github.com/n0-computer/iroh/pull/1971 btw.  Let's merge main into this branch here once that's merged.

**Pull request merged**